### PR TITLE
Hierarchical regions updates

### PR DESCRIPTION
**Pull request by rebecca-caktus**
With a few other things mixed in.

* Prevent flash of un-nested content by adding "Loading..." text.
* Fix drag & drop style issues.
* Fix bug when saving regions. Some changes cause the tree to get out of sync during partial rebuilds that occur during `delay_mptt_updates`. This uses `disable_mptt_updates` to avoid all rebuilding until all changes are made.
* Downgrade `pep8` and `flake8` to versions supported by `flake8`. I had noticed "flakey" behavior and discovered this as the cause.
* Update Bootstrap to 3.3.5. While playing with glyphicons I noticed some bugs that have been fixed in the more recent version of Bootstrap. I created a [pull request](https://github.com/nyaruka/smartmin/pull/61) to update smartmin.
* Beginning preparations for possibility of custom user model.

### CODE REVIEW

File: tracpro/groups/views.py [hunk 1]
```diff
@@ -108,26 +108,31 @@ def post(self, request, *args, **kwargs):
...
-                       "or null. ")
-                logger.warning("Hierarchy: " + msg + raw_data)
+                       "or null.")
+                logger.warning("{} Hierarchy: {} {}".format(org, msg, raw_data))
                 return self.error_response(400, msg)
 
-            # Re-set parent values for each region.
-            with Region.objects.delay_mptt_updates():
+            # Re-set parent values for each region, then rebuild the mptt tree.
+            with Region.objects.disable_mptt_updates():
```
Comment (emullaney) [inline tracpro/groups/views.py]: Have you tried wrapping this inside of a database transaction? The mptt docs suggest you do this.
https://django-mptt.github.io/django-mptt/mptt.managers.html#mptt.managers.TreeManager.disable_mptt_updates

Comment (rebecca-caktus) [inline tracpro/groups/views.py]: I am using the `@transaction.atomic` decorator on the method.

Comment (emullaney) [inline tracpro/groups/views.py]: Nice! I missed that, should have scrolled up.

File: tracpro/groups/views.py [hunk 2]
```diff
@@ -108,26 +108,31 @@ def post(self, request, *args, **kwargs):
...
                 for region_id, parent_id in data.items():
                     region = regions.get(str(region_id))
                     parent = regions.get(str(parent_id))
                     if region.parent != parent:
+                        old = region.parent.name if region.parent else None
+                        new = parent.name if parent else None
+                        msg = "Updating parent of {} from {} -> {}".format(region, old, new)
```
Comment (emullaney) [inline tracpro/groups/views.py]: Nice detail on this log message.

**Comment by emullaney** [comment]
:palm_tree:

**Pull request merged**